src/chunk.c

```diff
@@ -2739,12 +2740,14 @@ ts_chunk_has_associated_compressed_chunk(int32 chunk_id)
 	ts_scanner_foreach(&iterator)
 	{
 		TupleInfo *ti = ts_scan_iterator_tuple_info(&iterator);
-		bool isnull;
-		heap_getattr(ti->tuple, Anum_chunk_compressed_chunk_id, ti->desc, &isnull);
-		compressed = !isnull; // isnull is false when compress chunk id is set
+		bool chunk_id_isnull, dropped_isnull;
+		Datum dropped;
+		heap_getattr(ti->tuple, Anum_chunk_compressed_chunk_id, ti->desc, &chunk_id_isnull);
+		dropped = heap_getattr(ti->tuple, Anum_chunk_dropped, ti->desc, &dropped_isnull);
+		can_be_compressed = !chunk_id_isnull && !dropped_isnull && !DatumGetBool(dropped);
```

Comment:
`dropped_isnull` cannot be true since there is a `NOT NULL` constraint on this attribute. I think we should instead `Assert(!dropped_isnull);`

Reply:
Done.